### PR TITLE
[DONE] Added support for token auth

### PR DESCRIPTION
**Pull request by jpprins1**
Access/refresh token support for 'ThreediAPI' class. (used in Threedi-cmd)

### CODE REVIEW

File: threedi_api_client/api.py [hunk 1]
```diff
@@ -85,14 +85,32 @@ def __init__(self, env_file=None, config=None, version="v3", asynchronous=False)
...
                 "ThreediApi requires the THREEDI_API_HOST, THREEDI_API_USERNAME, "
-                "and THREEDI_API_PASSWORD configuration values."
+                "configuration values."
+            )
+
+        # Get the config variables
+        password = user_config.get("THREEDI_API_PASSWORD")
+        try:
+            access_token = user_config.get("THREEDI_API_ACCESS_TOKEN")
+        except KeyError:
```
Comment (caspervdw) [inline threedi_api_client/api.py:100]: It is unexpected that the `.get` would raise a `KeyError`, this is not like a dict. I made a ticket to refactor this a bit: https://github.com/nens/threedi-api-client/issues/56

**Review by caspervdw — CHANGES_REQUESTED** [review]
This looks good, only there are two things missing:

- unittests should be added for the new 'get config variables' logic (say, one with host + username + tokens, and one with only 1 token)

File: threedi_api_client/api.py [hunk 2]
```diff
@@ -85,14 +85,32 @@ def __init__(self, env_file=None, config=None, version="v3", asynchronous=False)
...
+            access_token = None
+        try:
+            refresh_token = user_config.get("THREEDI_API_REFRESH_TOKEN")
+        except KeyError:
+            refresh_token = None
+
+        tokens = all(x for x in (access_token, refresh_token))
```
Comment (caspervdw) [inline threedi_api_client/api.py:107]: Better to be more strict in user-facing API.

If you only supply one access and not a refresh token, raise that they both have to be given.
Also, if you supply both tokens and password, raise that you have to choose between one of the two authentication options.

File: threedi_api_client/api.py (hunk 1 above)
Comment (jpprins1) [reply threedi_api_client/api.py:100]: Okay, I'll perform the proposed refactor in this PR as well.

**Pull request merged**